### ISSUE TITLE
Pitch accent

### ISSUE THREAD

**Issue by Marcusjmdict** [JMdictProject/JMdictIssues#42]
I thought we might as well have a thread on the implementation of pitch accent here? I'm also sort of wondering how far along it is and/or if there's a road map as to when it will eventually be implemented.

**Comment by JMdictProject** [comment]
Yes, a good idea to have an issue on this. The development timetable for the database itself is a separate issue and best handled elsewhere. The bare-bones outline is at:  http://www.edrdg.org/wiki/index.php/JMdict:_Next_Generation#Pitch_Accent_Elements

The topic got some discussion on the mailing list - see https://www.edrdg.org/jmdict_edict_list/2020/threads.html  near the top. Some of that discussion could possibly be copied into here.

**Comment by Marcusjmdict** [comment]
The development timetable (as it pertains to JMdict NG?) might be better discussed elsewhere but I do want to stress that this is a feature I think we should really try and implement sooner rather than later. Our lack of this data is a source of frequent complaints and several apps/sites have begun compiling this data by themselves from other sources (the "Kanjium" project for example is used by Yomichan: https://community.wanikani.com/t/psa-yomichan-now-has-built-in-pitch-accent-support/43259 )
It's a great shame we can't provide this natively to developers yet but that they must jump through hoops to provide this to the end users (despite broad agreement on the mailing list already in April 2019 that this should be implemented in JMdict). This is a volunteer project for all involved of course but I do feel that this deserves to be prioritized.

As with my xref suggestion, I'd really like to see this implemented as soon as possible, preferably ahead of JMdict NG.

**Comment by spectria-limina** [comment]
Random language learner here, poking the issue to see if there's signs of life. I don't have the language expertise to confidently make any statements about accent, but I can at least summarize the most salient messages from the mailing list, and offer some of my own random devbrain thoughts:

* Melissa Boiko notes the difference between pitch (which is present on every mora) and accent (which, in Tokyo Japanese, is the mora, if any, where the downstep occurs). In Tokyo Japanese, the accent fully describes the pitch of all mora, which would be a simpler data model. Other dialects could have similar simplified models. A full phonetic description of high/low needs to be able to additionally include the trailing particle. Phonetic might be easier for app makers.
* Jim Breen mentioned only initially handling Tokyo accent, and that data availability could be an issue.
* Darren Cook asked if the differences between dialects is fairly algorithmic, or largely exceptional, as well as how much JMDict actually wants to include if comprehensive data were available.
* Melissa explained that dialectal pitch accent is fairly predictable diachronically, albeit with a lot of exceptions and variation, starting from Middle Japanese, but unpredictable synchronically.
* Alan pointed out that a single word, even in Tokyo Japanese, may have multiple pitch accents, e.g. 薙刀
* Stuart McGraw added that there may be different pitch accents for different meanings of the same word, such as クラブ which is different if it means "club" or "crab". He also pointed out that many terms are compounds with multiple words. In such cases then in order to derive the pitch contour, in addition to the accented morae for each word, the word boundaries would need to be known.
* Jim pointed out this case would be argued to be a case of homographs whose entry should be split. But this doesn't hold up for e.g. 一番 whose accent changes based on sense/part of speech of the particular word.
    * I would add that [onomatopoeia](https://japanese.stackexchange.com/questions/61921/pitch-accent-variations-within-giongo-words-like-%e3%81%b5%e3%82%8f%e3%81%b5%e3%82%8f#comment107482_61921) are the same.
* Melissa pointed out more rules that would beed to be applied when trying to derive pitch contours for compounds, and the infeasibility of doing so. For multi-accent compounds, therefore, just the contour should be stored.

My summary of design & suggestions in light of the above:
* The primary goal is Tokyo Japanese pitch accent, with dialects as future work but that should be futureproofed for.
     * IMO the "correct" approach is an `xml:lang` attribute, although there are no standard dialect tags so it would require defining private-use tags (e.g. `ja-JP-x-kansai`). Alternatively a custom "dialect" attribute might be more sensible and less repetitive.
* Support for both pitch countour (phonetic) and accent (phonemic) info. For other dialects, the phonemic info could be supplemented with relevant information, but that is well out of scope for a first version of pitch accent.
    * A minimal syntax for contour, like `LHHLL`, seems good. For the particle, Melissa's approach of separating the particle from the rest of the word with \*, e.g. `LHHH*H` seems fine enough.
    * Pitch contour alone would be sufficient, but it seems to me there is no reason not to make the number of the accented mora also available, for simple compounds.
    * Making the contour optional might facilitate data import, although there's a risk that if it is always present, users might code an assumption and reject a later datafile that omits it. So perhaps contour should be required for every accent.
* The relationship of accent to sense is many-to-many, and this requires a way to represent it.
    * I think the best way would be to reuse the mechanism of the current `re_restr`, e.g. `re_pa_rstr`, which contains the numerical index of a sense. If no `re_pa_rstr` is present, the accent pattern applies to all senses.
    * There should probably be a usage notes field for a pitch accent pattern as well.
